pkg/client/restclient/request.go

```diff
@@ -548,6 +549,24 @@ func (r *Request) decoder(contentType string) (runtime.Decoder, bool) {
 	return negotiated.DecoderToVersion(serializer, internalGV), true
 }
 
+func (r *Request) streamingDecoder(contentType string, reader io.ReadCloser) (streaming.Decoder, runtime.Decoder, bool) {
```

Comment:
Although this is better than doing it up front (since most clients won't watch) I do think we should memoize this some how.  We need to get a codec if one is not set, but we don't really support dynamically changing content-type afterwards.  I guess we have a few cases where we might want to speculatively use one type, and then fall back to another, but it's rare.